targets/TARGET_NUVOTON/TARGET_M2351/TARGET_NUMAKER_PFM_M2351/TARGET_M23_NS/NuMaker-mbed-TZ-secure-example.hex

```diff
@@ -1,797 +1,873 @@
 :020000040000FA
-:1000000000080020050F0000D90B0000D90B0000EC
+:1000000000080020AB100000190D0000190D0000C1
```

Comment:
Should this file and also another object file in the same folder contain a license file? (https://github.com/ARMmbed/mbed-os/tree/a65e3d42e53e5c46c8a5bd74f17c5d0e7cef0f1b/targets/TARGET_NUVOTON/TARGET_M2351/TARGET_NUMAKER_PFM_M2351/TARGET_M23_NS)

Would be good to have it as a separate PR

Reply:
@0xc0170 The [Permissive Binary License](https://www.mbed.com/en/licenses/permissive-binary-license/) meets the scenario. Could we reuse this license just like it is attached in https://github.com/ARMmbed/mbed-os/tree/master/tools/bootloaders/MTB_MTS_DRAGONFLY?
@cyliangtw

Reply:
Yes.

Regarding this hex file - what does it do as it has suffix -example ?

Reply:
@0xc0170 Thanks. 

> Regarding this hex file - what does it do as it has suffix -example ?

TrustZone target requires two codes, secure and non-secure. Secure code would run first to set up security environment and then bring up non-secure code. **NuMaker-mbed-TZ-secure-example.hex** is the default secure code (and **cmse_lib.o** is the secure gateway library exported by it). It is built from:
https://github.com/OpenNuvoton/NuMaker-mbed-TZ-secure-example
We choose **.hex** rather than **.bin** as output format because secure/non-secure code may not start at address `0x0`. To flash secure/non-secure code on **M2351**, we would drag-n-drop **NuMaker-mbed-TZ-secure-example.hex** first and then user program e.g. **mbed-os-example-blinky.hex**.